src/rgw/rgw_user.cc

```diff
@@ -1850,6 +1861,12 @@ int RGWUser::check_op(RGWUserAdminOpState& op_state, std::string *err_msg)
     return -EINVAL;
   }
 
+  if (rgw_is_valid_tenant(op_id.tenant)) {
```

Comment:
not sure exactly what you mean here, but i think this should be:
```c++
  ret = rgw_is_valid_tenant(op_id.tenant);
  if (ret < 0) {
    set_err_msg(err_msg, ...);
    return ret;
  }
```
past that, it doesn't look like rgw_admin.cc or rgw_rest_user.cc need to do anything special?